lib/evmone/analysis.cpp

```diff
@@ -10,7 +10,7 @@ namespace evmone
 {
 struct block_analysis
 {
-    int gas_cost = 0;
+    int64_t gas_cost = 0;
```

Comment:
I think I see what's the point, it's just a bit confusing that this is signed while `block_info::gas_cost` is unsigned. Does it worth changing this to `uint64_t`?

Reply:
Unfortunately, the signed type is better here. It is an error (detectable with undefined behavior sanitizer) to overflow signed types while unsigned have defined behavior on overflow.

The good way of thinking here is as follows: we use `int64_t` for gas computation (here and during execution), and `uint32_t` is used only for storing block gas cost.